src/decoder_fmmidi.cpp

```diff
@@ -66,22 +68,39 @@ bool FmMidiDecoder::Open(FILE* file) {
 	}
 	seq->rewind();
 
-	ticks_per_sec = (float)seq->get_division() / tempo * 1000000;
+	tempo.emplace_back(this, midi_default_tempo);
 
 	return true;
 }
 
 bool FmMidiDecoder::Seek(size_t offset, Origin origin) {
+	assert(!tempo.empty());
+
 	if (offset == 0 && origin == Origin::Begin) {
 		mtime = seq->rewind_to_loop();
-		mtime_last_tempo_change = mtime_ltc_loopstart;
-		ticks_last_tempo_change = ticks_ltc_loopstart;
+
 		if (mtime > 0.0f) {
+			// Throw away all tempo data after the loop point
+			auto it = tempo.end();
+			while (it > tempo.begin()) {
+				--it;
+				if (it->mtime <= mtime) {
+					++it;
+					break;
+				}
+			}
+			if (it != tempo.end()) {
+				tempo.erase(it, tempo.end());
```

Comment:
I believe this whole loop can be:
```
auto riter = std::find_if(tempo.rbegin(), tempo.rend(), [&](auto& t) { return t .mtime <= seq->rewind_to_loop; });
auto iter = std::next(riter).base();
if (iter != tempo.end()) {
   tempo.erase(iter);
}
```

I've never tried the `base()` trick with reverse iterator before.  So test before you try this.